suuchi-core/src/main/scala/in/ashwanthkumar/suuchi/store/VersionedStore.scala

```diff
@@ -5,20 +5,38 @@ import scala.util.hashing.MurmurHash3
 
 
 object Versions {
-  def fromBytes(bytes: Array[Byte]) = fromString(new String(bytes))
-  def fromString(versionString: String) = versionString.split('|').map(_.toLong)
-  def toString(versions: List[Long]) = versions.map(_.toString).mkString("|")
-  def toBytes(versions: List[Long]) = toString(versions).getBytes
+  /*
+  *
+  * Serialization util for managing list of Versions of type Long
+  * We follow the below protocol to serialize List[Long] to Array[Byte]
+  *
+  * First byte - # of elements in the list - N
+  * We will have N "longs represented as bytes - 8 bytes each" following it
+  * Since, we know each version is a long, we just read 8 bytes to construct a long and move forward.
+  * */
+
+  def fromBytes(bytes: Array[Byte]): List[Long] = {
+    val numVersions = bytes(0).toInt
+    bytes.drop(1).sliding(8, 8).map(PrimitivesSerDeUtils.bytesToLong) toList
+  }
+
+  def toBytes(versions: List[Long]): Array[Byte] = {
+    val numVersions = versions.size.toByte
```

Comment:
Using a `byte` limits the number of versions we support isn't? That's intentional? Int is long enough generally but `Byte` feels so small. Thoughts?